types/messages.go

```diff
@@ -138,103 +130,206 @@ func (msg *SSVMessage) Decode(data []byte) error {
+	if msg.SSVMessage == nil {
+		return errors.New("nil SSVMessage")
+	}
+
 	return nil
 }
 
-// EncodeSignedSSVMessage serializes the message, op id and signature into bytes
-func EncodeSignedSSVMessage(message []byte, operatorID OperatorID, signature [256]byte) []byte {
-	b := make([]byte, signatureSize+operatorIDSize+len(message))
-	copy(b[signatureOffset:], signature[:])
-	binary.LittleEndian.PutUint64(b[operatorIDOffset:], operatorID)
-	copy(b[messageOffset:], message)
-	return b
+func SSVMessageToSignedSSVMessage(msg *SSVMessage, operatorID OperatorID, signSSVMessageF SignSSVMessageF) (*SignedSSVMessage, error) {
+
+	sig, err := signSSVMessageF(msg)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not sign SSVMessage")
+	}
+
+	return &SignedSSVMessage{
+		Signatures:  [][]byte{sig},
+		OperatorIDs: []OperatorID{operatorID},
+		SSVMessage:  msg,
+	}, nil
 }
 
-// DecodeSignedSSVMessage deserializes signed message bytes messsage, op id and a signature
-func DecodeSignedSSVMessage(encoded []byte) ([]byte, OperatorID, [256]byte, error) {
-	if len(encoded) < messageOffset {
-		return nil, 0, [256]byte{}, fmt.Errorf("unexpected encoded message size of %d", len(encoded))
+// DeepCopy returns a new instance of SignedMessage, deep copied
+func (signedMsg *SignedSSVMessage) DeepCopy() *SignedSSVMessage {
+	ret := &SignedSSVMessage{
+		OperatorIDs: make([]OperatorID, len(signedMsg.GetOperatorIDs())),
+		Signatures:  make([][]byte, len(signedMsg.Signatures)),
+	}
+	copy(ret.OperatorIDs, signedMsg.GetOperatorIDs())
+	copy(ret.Signatures, signedMsg.Signatures)
+
+	retSSV := &SSVMessage{
+		MsgType: signedMsg.SSVMessage.MsgType,
+		Data:    make([]byte, len(signedMsg.SSVMessage.Data)),
 	}
+	msgID := [56]byte{}
+	copy(msgID[:], signedMsg.SSVMessage.MsgID[:])
+	retSSV.MsgID = msgID
 
-	message := encoded[messageOffset:]
-	operatorID := binary.LittleEndian.Uint64(encoded[operatorIDOffset : operatorIDOffset+operatorIDSize])
-	signature := [256]byte{}
-	copy(signature[:], encoded[signatureOffset:signatureOffset+signatureSize])
-	return message, operatorID, signature, nil
+	copy(retSSV.Data, signedMsg.SSVMessage.Data)
+
+	if len(signedMsg.FullData) > 0 {
+		ret.FullData = make([]byte, len(signedMsg.FullData))
+		copy(ret.FullData, signedMsg.FullData)
+	}
+	ret.SSVMessage = retSSV
+	return ret
 }
 
-func SSVMessageToSignedSSVMessage(msg *SSVMessage, operatorID OperatorID, signSSVMessageF SignSSVMessageF) (*SignedSSVMessage, error) {
-	encodedSSVMsg, err := msg.Encode()
-	if err != nil {
-		return nil, errors.Wrap(err, "could not encode SSVMessage")
+// MatchedSigners returns true if the provided signer ids are equal to GetOperatorIDs() without order significance
+func (msg *SignedSSVMessage) MatchedSigners(ids []OperatorID) bool {
+	if len(msg.GetOperatorIDs()) != len(ids) {
+		return false
+	}
+
+	for _, id := range msg.GetOperatorIDs() {
+		found := false
+		for _, id2 := range ids {
+			if id == id2 {
+				found = true
+			}
+		}
+
```

Comment:
could probably used slices.Contains

Reply:
ok @MatheusFranco99 
I think I found some bugs here:
1. Your comment talks about equality: `// MatchedSigners returns true if the provided signer ids are equal to GetOperatorIDs() without order significance`. To check equality you need to do two way containment test, not one (maybe there is a util function). But I think the bug is in the comment and not in the logic.
2. I think you wanted to do a one way containment test, but I think it should be the other way? That is because from `func (c *MsgContainer) AddFirstMsgForSignerAndRound` I understand we need to check whether all `ids` are included in `msg`?

Reply:
@MatheusFranco99 pointed out that this is the same logic as before

So opened #387 
Later we inspect and fix in a separate PR